docs/docsite/rst/network/dev_guide/developing_plugins_network.rst

```diff
@@ -0,0 +1,108 @@
+
+Each network connection plugin has a set of plugins which provide a specification of the connection for a particular set of devices. Public methods of these plugins may be called on the connection proxy object from the module just as connection methods can.
+
+.. code-block:: python
+
+  from ansible.module_utils.connection import Connection
+
+  ...
+
+  # module is your AnsibleModule instance
+  connection = Connection(module._socket_path)
+
+  # You can now call any method (that doesn't start with '_') of the connection
+  # plugin or its platform-specific plugin
+  connection.get_config()
+
+.. contents::
+   :local:
+
+.. _developing_plugins_httpapi:
+
+Developing httpapi plugins
+==========================
+
+:ref:`httpapi plugins <httpapi_plugins>` serve as adapters for various HTTP(S) APIs for use with the ``httpapi`` connection plugin. They should implement a minimal set of convenience methods tailored to the API you are attempting to use.
+
+Specifically, there are a few methods that the ``httpapi`` connection plugin expects to exist.
+
+Making requests
+---------------
+
+The ``httpapi`` connection plugin has a ``send()`` method, but an httpapi plugin needs a ``send_request(self, data, **message_kwargs)`` method as a higher-level wrapper to ``send()``. This method should prepare requests by adding fixed values like common headers or URL root paths. This method may do more complex work such as turning data into formatted payloads, or determining which path or method to request. It may then also unpack responses to be more easily consumed by the caller.
+
+.. code-block:: python
+
+   def send_request(self, data, path, method='POST'):
+       # Fixed headers for requests
+       headers = {'Content-Type': 'application/json'}
+       try:
+           response, response_content = self.connection.send(path, data, method=method, headers=headers)
+       except HTTPError as exc:
+           return exc.code, exc.read()
+
+       # handle_response (defined separately) will take the format returned by the device
+       # and transform it into something more suitable for use by modules.
+       # This may be JSON text to Python dictionaries, for example.
+       return handle_response(response_content)
+
+Authenticating
+--------------
+
+By default, all requests will authenticate with HTTP Basic authentication. If a request can return some kind of token to stand in place of HTTP Basic, the ``update_auth(self, response, response_text)`` method should be implemented to inspect responses for such tokens. If the token is meant to be included with the headers of each request, it is sufficient to return a dictionary which will be merged with the computed headers for each request. The default implementation of this method does exactly this for cookies. If the token is used in another way, say in a query string, you should instead save that token to an instance variable, where the ``send_request()`` method (above) can add it to each request
```

Comment:
Based solely on this paragraph, it's not clear to me how to structure calls which use a static header token and don't need any login or logout calls. The paragraph opens with "By default..." so by just reading this paragraph, I'm concerned I will need to disable that.

Reply:
I'm not entirely following your question. This is the section that deals with not having login or logout calls. If your header token is already known beforehand and there is no other authentication, then you would have to take some other approach not listed here (probably adding the token as a plugin option, and setting `self.connection._auth` to use the value of that option in `__init__` similar to how the `login()` example does it below)

Reply:
I think that makes sense. I'll need to test my implementation out a little more thoroughly to be sure, but it does mostly answer my question. I wouldn't mind seeing a note stating this as such though.